### PR TITLE
Fix/#1119 tx invalid

### LINKED ISSUE #1119: Txs set as invalid in the pool
[json-rpc.log](https://github.com/0xPolygonHermez/zkevm-node/files/9487301/json-rpc.log)
[prover.log](https://github.com/0xPolygonHermez/zkevm-node/files/9487302/prover.log)
[sequencer.log](https://github.com/0xPolygonHermez/zkevm-node/files/9487303/sequencer.log)
pool.txs and L2Block table:
[l2block_202209050845.csv](https://github.com/0xPolygonHermez/zkevm-node/files/9487374/l2block_202209050845.csv)
[txs_202209050844.csv](https://github.com/0xPolygonHermez/zkevm-node/files/9487375/txs_202209050844.csv)

Three txs were marked as invalid in the pool with no reason. They should be processed. The account had balance and the nonces were correct.
Interesting log from the sequencer (full log attached):
```
2022-09-05T06:20:14Z	INFO	state/state.go:1059	Tx 0xaa29a4e5fd72ef3afd4b51650a6a5d62d2270f55a7bd88bfcdc66a8c7d90ab60 has not been processed
2022-09-05T06:20:14Z	INFO	state/state.go:1059	Tx 0x9d64b342e7449368aebf81ee1b383433261076b759c0927c5329cbe28ccf8535 has not been processed
2022-09-05T06:20:14Z	ERROR	sequencer/batchbuilder.go:206	failed to store transactions, err: there are more transactions in the database than in the processed transaction set%!(EXTRA string=
/src/log/log.go:106 github.com/0xPolygonHermez/zkevm-node/log.appendStackTraceMaybeArgs()
/src/log/log.go:164 github.com/0xPolygonHermez/zkevm-node/log.Errorf()
/src/sequencer/batchbuilder.go:206 github.com/0xPolygonHermez/zkevm-node/sequencer.(*Sequencer).storeProcessedTransactions()
/src/sequencer/batchbuilder.go:64 github.com/0xPolygonHermez/zkevm-node/sequencer.(*Sequencer).tryToProcessTx()
/src/sequencer/sequencer.go:102 github.com/0xPolygonHermez/zkevm-node/sequencer.(*Sequencer).Start.func1()
)
```

This log is from other service but here you can see the txs sent and the order:
```
2022-09-05T08:15:41+02:00       INFO    cmd/main.go:265 Address: 0x70997970C51812dc3A010C7d01b50e0d17dc79C8
2022-09-05T08:15:41+02:00       INFO    cmd/main.go:158 chainID: 1000
2022-09-05T08:15:41+02:00       DEBUG   cmd/main.go:159 interval: 30s
2022-09-05T08:16:11+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:2 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:16:11+02:00       INFO    cmd/main.go:223 Tx hash: 0xf0f401270c1238115593d94a19c93199455ccfe4d9aa9e756d92e361ddea56d8
2022-09-05T08:16:11+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:3 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:16:11+02:00       INFO    cmd/main.go:223 Tx hash: 0x8480a02e5fe3e80a2338c24efdc777d0a0fd47cdd864d7eaedde260b1a64d8e2
2022-09-05T08:16:11+02:00       INFO    cmd/main.go:231 Txs sent!
2022-09-05T08:16:41+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:4 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:16:41+02:00       INFO    cmd/main.go:223 Tx hash: 0xb0e7c9fdfa77a8f42b9594e4990dda08b8fc01d8ba43d58679de3f2bdfa956b6
2022-09-05T08:16:41+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:5 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:16:41+02:00       INFO    cmd/main.go:223 Tx hash: 0xfae2203663c12f30fb575c31a27820734cfa4791407998379f2baae61623d3e0
2022-09-05T08:16:41+02:00       INFO    cmd/main.go:231 Txs sent!
2022-09-05T08:17:11+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:6 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:17:11+02:00       INFO    cmd/main.go:223 Tx hash: 0x8b070cb1897c85cf873078c78e1e30536a2f8ae153b1876e4869adbc94ffb7df
2022-09-05T08:17:12+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:7 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:17:12+02:00       INFO    cmd/main.go:223 Tx hash: 0x3fcd03d4ba5faae75263e6b817c0f4e12812d8da61d438576871c1d2a50ab998
2022-09-05T08:17:12+02:00       INFO    cmd/main.go:231 Txs sent!
2022-09-05T08:17:42+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:8 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:17:42+02:00       INFO    cmd/main.go:223 Tx hash: 0xd49d6611a285f61ccacb10bfeaecafa0afb7d82548e05fc12c1924f5bf871365
2022-09-05T08:17:42+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:9 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:17:42+02:00       INFO    cmd/main.go:223 Tx hash: 0xaa15f23eefdc36c241db05e88dbc089cc650f89a8a48b8cc4706619faedb59e9
2022-09-05T08:17:42+02:00       INFO    cmd/main.go:231 Txs sent!
2022-09-05T08:18:12+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:10 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:18:12+02:00       INFO    cmd/main.go:223 Tx hash: 0x1134f2e65df06fb504b03cdd0583729cb2e1d44ced1b95164d37f57c91335277
2022-09-05T08:18:12+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:11 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:18:12+02:00       INFO    cmd/main.go:223 Tx hash: 0x0e642a87290695e3aa7011b854ca569d82b4f6a45cf7546ae49c722f8d6f815c
2022-09-05T08:18:12+02:00       INFO    cmd/main.go:231 Txs sent!
2022-09-05T08:18:42+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:12 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:18:42+02:00       INFO    cmd/main.go:223 Tx hash: 0x5e441983cdd9c6c9c36ff16269e6bef2c2ed5ecd786065003dfc7698510c467d
2022-09-05T08:18:42+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:13 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:18:42+02:00       INFO    cmd/main.go:223 Tx hash: 0x771d4073c536b577382ab11c65b02b64a1af23f156bc1c98431d36e9c508b7ba
2022-09-05T08:18:42+02:00       INFO    cmd/main.go:231 Txs sent!
2022-09-05T08:19:12+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:14 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:19:12+02:00       INFO    cmd/main.go:223 Tx hash: 0x9680ed2c56fc9dd3e96e1877b197ba8c93e04eebd85d1e5d7782edbcf6889368
2022-09-05T08:19:12+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:15 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:19:12+02:00       INFO    cmd/main.go:223 Tx hash: 0xe96d8a4d3dbea9e0f15fe282d60b6710a4281a4912e1cf73e6d1ade6f278fdf0
2022-09-05T08:19:12+02:00       INFO    cmd/main.go:231 Txs sent!
2022-09-05T08:19:42+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:16 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:19:42+02:00       INFO    cmd/main.go:223 Tx hash: 0xaa29a4e5fd72ef3afd4b51650a6a5d62d2270f55a7bd88bfcdc66a8c7d90ab60
2022-09-05T08:19:42+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:17 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:19:42+02:00       INFO    cmd/main.go:223 Tx hash: 0x0363d94e0682c158bc66aaeff40e94ab44fdc65ebdf87d4c8c1e8c2725c9c5e8
2022-09-05T08:19:42+02:00       INFO    cmd/main.go:231 Txs sent!
2022-09-05T08:20:12+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:18 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:20:12+02:00       INFO    cmd/main.go:223 Tx hash: 0x9d64b342e7449368aebf81ee1b383433261076b759c0927c5329cbe28ccf8535
2022-09-05T08:20:12+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:19 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:20:12+02:00       INFO    cmd/main.go:223 Tx hash: 0xd94a2ccede157afadcd18489c9ed5d585c7d783192993fb51927b2fb28a0f2b2
2022-09-05T08:20:12+02:00       INFO    cmd/main.go:231 Txs sent!
2022-09-05T08:20:42+02:00       DEBUG   cmd/main.go:217 Tx: &{Nonce:17 GasPrice:+1000000000 Gas:21000 To:0x2536C2745Ac4A584656A830f7bdCd329c94e8F30 Value:+10000000000000000 Data:[] V:<nil> R:<nil> S:<nil>}
2022-09-05T08:20:42+02:00       INFO    cmd/main.go:223 Tx hash: 0x0363d94e0682c158bc66aaeff40e94ab44fdc65ebdf87d4c8c1e8c2725c9c5e8
2022-09-05T08:20:42+02:00       ERROR   cmd/main.go:226 error sending tx. Error: already known
/home/rr/Documentos/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-node@v0.0.0-20220831160929-d7ff1cb69115/log/log.go:106 github.com/0xPolygonHermez/zkevm-node/log.appendStackTraceMaybeArgs()
/home/rr/Documentos/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-node@v0.0.0-20220831160929-d7ff1cb69115/log/log.go:131 github.com/0xPolygonHermez/zkevm-node/log.Error()
/home/rr/Documentos/Iden3/zkevm-continuity-testing/cmd/main.go:226 main.sendEthTxs()
/home/rr/Documentos/Iden3/zkevm-continuity-testing/cmd/main.go:161 main.start()
/home/rr/Documentos/go/pkg/mod/github.com/urfave/cli/v2@v2.14.0/command.go:173 github.com/urfave/cli/v2.(*Command).Run()
/home/rr/Documentos/go/pkg/mod/github.com/urfave/cli/v2@v2.14.0/app.go:384 github.com/urfave/cli/v2.(*App).RunContext()
/home/rr/Documentos/go/pkg/mod/github.com/urfave/cli/v2@v2.14.0/app.go:253 github.com/urfave/cli/v2.(*App).Run()
/home/rr/Documentos/Iden3/zkevm-continuity-testing/cmd/main.go:102 main.main()
/usr/local/go/src/runtime/proc.go:250 runtime.main()

2022-09-05T08:20:42+02:00       FATAL   cmd/main.go:104 error sending eth txs. Error: already known
/home/rr/Documentos/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-node@v0.0.0-20220831160929-d7ff1cb69115/log/log.go:106 github.com/0xPolygonHermez/zkevm-node/log.appendStackTraceMaybeArgs()
/home/rr/Documentos/go/pkg/mod/github.com/0x!polygon!hermez/zkevm-node@v0.0.0-20220831160929-d7ff1cb69115/log/log.go:137 github.com/0xPolygonHermez/zkevm-node/log.Fatal()
/home/rr/Documentos/Iden3/zkevm-continuity-testing/cmd/main.go:104 main.main()
/usr/local/go/src/runtime/proc.go:250 runtime.main()

exit status 1
```

### PR DESCRIPTION
**Pull request by ARR552**
Closes #1119 

### What does this PR do?

This PR fixes the invalid txs in the pool.

### Reviewers

- @Mikelle 
- @arnaubennassar 
- @KonradIT

### CODE REVIEW

File: sequencer/batchbuilder.go [hunk 1]
```diff
@@ -34,10 +34,10 @@ func (s *Sequencer) tryToProcessTx(ctx context.Context, ticker *time.Ticker) {
 			waitTick(ctx, ticker)
 			return
 		} else if err != nil {
-			log.Errorf("error closing sequence: %v", err)
+			log.Errorf("error closing sequence: %s", err.Error())
```
Comment (arnaubennassar) [inline sequencer/batchbuilder.go:37]:
```suggestion
			log.Errorf("error closing sequence: %w", err)
```

File: sequencer/batchbuilder.go [hunk 2]
```diff
@@ -34,10 +34,10 @@ func (s *Sequencer) tryToProcessTx(ctx context.Context, ticker *time.Ticker) {
...
 			log.Info("resetting sequence in progress")
 			if err = s.loadSequenceFromState(ctx); err != nil {
-				log.Error("error loading sequence from state: %v", err)
+				log.Error("error loading sequence from state: %s", err.Error())
```
Comment (arnaubennassar) [inline sequencer/batchbuilder.go:40]:
```suggestion
				log.Error("error loading sequence from state: %w", err)
```

File: sequencer/batchbuilder.go [hunk 3]
```diff
@@ -78,7 +78,7 @@ func (s *Sequencer) newSequence(ctx context.Context) (types.Sequence, error) {
 	if s.lastStateRoot.String() != "" || s.lastLocalExitRoot.String() != "" {
 		dbTx, err = s.state.BeginStateTransaction(ctx)
 		if err != nil {
-			return types.Sequence{}, fmt.Errorf("failed to begin state transaction to close batch, err: %v", err)
+			return types.Sequence{}, fmt.Errorf("failed to begin state transaction to close batch, err: %s", err.Error())
```
Comment (arnaubennassar) [inline sequencer/batchbuilder.go:81]:
```suggestion
			return types.Sequence{}, fmt.Errorf("failed to begin state transaction to close batch, err: %w", err)
```

File: sequencer/batchbuilder.go [hunk 4]
```diff
@@ -121,7 +121,7 @@ func (s *Sequencer) calculateZkCounters() pool.ZkCounters {
 func (s *Sequencer) closeSequence(ctx context.Context) error {
 	newSequence, err := s.newSequence(ctx)
 	if err != nil {
-		return fmt.Errorf("failed to create new sequence, err: %v", err)
+		return fmt.Errorf("failed to create new sequence, err: %s", err.Error())
```
Comment (arnaubennassar) [inline sequencer/batchbuilder.go:124]: Same for all the changes

**Review by arnaubennassar — CHANGES_REQUESTED** [review]

**Review by Mikelle — APPROVED** [review]

**Review by arnaubennassar — APPROVED** [review]

**Pull request merged**